packages/transpiler/src/index.ts

```diff
@@ -89,22 +93,37 @@ export function transpileContracts(contracts: string[], artifacts: Artifact[]):
 
     const source = art.source;
 
-    const fileTran = fileTrans[art.fileName];
+    const fileTran = fileTrans[art.sourcePath];
     if (!fileTran.source) {
       fileTran.source = transpile(source, fileTran.transformations);
     }
-    const entry = acc.find(o => o.fileName === art.fileName);
+    const entry = acc.find(o => o.fileName === path.basename(art.sourcePath));
     if (!entry) {
-      const path = art.sourcePath.replace('.sol', 'Upgradable.sol');
-      let patchedFilePath = path;
-      if (path.startsWith('contracts')) {
-        patchedFilePath = path.replace('contracts/', '');
+      const upgradeablePath = path.normalize(art.sourcePath).replace('.sol', 'Upgradeable.sol');
+      let patchedFilePath = upgradeablePath;
+      // Truffle stores an absolute file path in a sourcePath of an artifact field
+      // "sourcePath": "/Users/iYalovoy/repo/openzeppelin-sdk/tests/cli/workdir/contracts/Samples.sol"
+      // OpenZeppelin stores relative paths
+      // "sourcePath": "contracts/Foo.sol"
+      // OpenZeppelin sourcePath would start with `contracts` for contracts present in the `contracts` folder of a project
+      // Both Truffle and OpenZeppelin support packages
+      // "sourcePath": "@openzeppelin/upgrades/contracts/Initializable.sol",
+      // Relative paths can only be specified using `.` and `..` for both compilers
```

Comment:
Do we have any tests to cover this?

Reply:
Only indirectly. SDK CLI integration tests use Truffle to build contracts and Transpiler to transpile them when Transpiler tests use OpenZeppelin compiler.